packages/compass-crud/src/components/insert-document-dialog.jsx

```diff
@@ -212,15 +212,17 @@ class InsertDocumentDialog extends React.PureComponent {
     const currentView = this.props.jsonView ? 'JSON' : 'List';
 
     return (
-      <ConfirmationModal
+      <FormModal
         title={`Insert to Collection ${this.props.ns}`}
         className="insert-document-dialog"
         open={this.props.isOpen}
-        onConfirm={this.handleInsert.bind(this)}
+        onSubmit={this.handleInsert.bind(this)}
         onCancel={this.props.closeInsertDocumentDialog}
-        buttonText="Insert"
+        submitButtonText="Insert"
         submitDisabled={this.hasErrors()}
         trackingId="insert_document_modal"
+        data-testid="insert-document-modal"
+        scrollClassName={css({ minHeight: spacing[6] * 2 })} // make sure there is enough space for the menu
```

Comment:
We could have a default minHeight for ModalContent, but then the ones that are designed to look like ConfirmationModal look weird.

But then without minHeight or overflow: 'visible' if you have a menu or similar popover thing that doesn't fit in the available area it will get cut off. Maybe there's a third option I'm missing?